### PR TITLE
Include keccak & aux lemmas conditionally

### PR DESCRIPTION
**Pull request by palinatolmach**
Closes https://github.com/runtimeverification/kontrol/issues/715.
Partially addresses https://github.com/runtimeverification/kontrol/issues/780.

This PR
- adds `keccak.md` file containing (unsound) assumptions about keccak to facilitate reasoning, use it by default
- adds `kontrol_lemmas.md` file containing other Kontrol lemmas that facilitate reasoning (copied from [Morpho lemmas](https://github.com/runtimeverification/_kaas_morpho/blob/c11b20e50ba694e9828fb6034bcbfca297aee278/test/kontrol/morpho-lemmas.k) with the exclusion of keccak assumptions included in `keccak.md`)
- includes both during `kontrol build` by default
- adds `--no-keccak-lemmas`, `--auxiliary-lemmas` options to `kontrol build` which excludes and includes keccak and auxiliary lemmas, respectively
- fixes a typo in an error message (`"provided"`)

### CODE REVIEW

File: src/kontrol/kompile.py [hunk 1]
```diff
@@ -57,7 +57,8 @@ def foundry_kompile(
         regen = True
         foundry_up_to_date = False
 
-    for r in options.requires:
+    requires = options.requires + (['src/kontrol/kdist/keccak.md'] if options.keccak_lemmas else [])
```
Comment (palinatolmach) [inline src/kontrol/kompile.py:60]: I forgot to make sure it's distributed with Kontrol, will add that too.

**Review by anvacaru — APPROVED** [review]

File: src/kontrol/kdist/kontrol_lemmas.md [hunk 2]
```diff
@@ -0,0 +1,466 @@
...
+    rule M:Map [ K1 <- _ ] [ K2 <- V2 ] [ K3 <- V3 ] [ K4 <- V4 ] [ K1 <- V5 ]                                        => M:Map [ K2 <- V2 ] [ K3 <- V3 ] [ K4 <- V4 ] [ K1 <- V5 ] [simplification]
+    rule M:Map [ K1 <- _ ] [ K2 <- V2 ] [ K3 <- V3 ] [ K4 <- V4 ] [ K5 <- V5 ] [ K1 <- V6 ]                           => M:Map [ K2 <- V2 ] [ K3 <- V3 ] [ K4 <- V4 ] [ K5 <- V5 ] [ K1 <- V6 ] [simplification]
+    rule M:Map [ K1 <- _ ] [ K2 <- V2 ] [ K3 <- V3 ] [ K4 <- V4 ] [ K5 <- V5 ] [ K6 <- V6 ] [ K1 <- V7 ]              => M:Map [ K2 <- V2 ] [ K3 <- V3 ] [ K4 <- V4 ] [ K5 <- V5 ] [ K6 <- V6 ] [ K1 <- V7 ] [simplification]
+    rule M:Map [ K1 <- _ ] [ K2 <- V2 ] [ K3 <- V3 ] [ K4 <- V4 ] [ K5 <- V5 ] [ K6 <- V6 ] [ K7 <- V7 ] [ K1 <- V8 ] => M:Map [ K2 <- V2 ] [ K3 <- V3 ] [ K4 <- V4 ] [ K5 <- V5 ] [ K6 <- V6 ] [ K7 <- V7 ] [ K1 <- V8 ] [simplification]
+
+    //
+    // Additional keccak assumption:
+    //
+
+    rule keccak(B:Bytes) <Int BOUND:Int => true requires BOUND >=Int pow256 -Int 32 [simplification, concrete(BOUND)]
```
Comment (anvacaru) [inline src/kontrol/kdist/kontrol_lemmas.md:422]: @palinatolmach should this be part of keccak.md?

Comment (palinatolmach) [reply src/kontrol/kdist/kontrol_lemmas.md:422]: I didn't find it among `keccak.md` lemmas, so left it there intentionally. But I can also move it to `keccak.md` or remove, if needed.

Comment (palinatolmach) [reply src/kontrol/kdist/kontrol_lemmas.md:422]: Oh, sorry, I misread your message. Yes, let me move it!

Comment (palinatolmach) [reply src/kontrol/kdist/kontrol_lemmas.md]: Done in https://github.com/runtimeverification/kontrol/pull/779/commits/cf2a4db2c1107c91451b46ac79db73c2f77ad4d5.

**Comment by palinatolmach** [comment]
Something's off with some integration tests but it doesn't happen locally — I'll try to re-run both in CI and locally too.

**Comment by PetarMax** [comment]
It seems to take too long, the task... perhaps it is better to not include by default, and include only on demand?

**Pull request merged**